src/dataproxy/worker/SharedWorkerProvider.tsx

```diff
@@ -41,13 +82,71 @@ export const SharedWorkerProvider = React.memo(
     const [tabCount, setTabCount] = useState(0)
     const client = useClient()
 
+    useEffect(() => {
+      if (!client) return
+
+      if (!hasRealtimePlugin(client.plugins)) {
+        throw new Error(
+          'You must include the realtime plugin to use RealTimeQueries'
+        )
+      }
+      const realtime = client.plugins.realtime
+
+      if (!realtime) {
+        throw new Error(
+          'You must include the realtime plugin to use RealTimeQueries'
+        )
+      }
+
+      const handleFileCreated = (event: unknown): void => {
+        if (!isFilesRealtimeEvent(event)) return
+        if (
+          event.dir_id === 'io.cozy.files.shared-drives-dir' &&
+          event.class === 'shortcut'
+        ) {
+          const driveId = event?.referenced_by?.[0]?.id
+          if (driveId) {
+            log.info(`Shared drive ${driveId} created`)
+            worker?.addSharedDrive(driveId)
+          }
+        }
+      }
+      const handleFileDeleted = (event: unknown): void => {
+        if (!isFilesRealtimeEvent(event)) return
+        if (
+          event.dir_id === 'io.cozy.files.shared-drives-dir' &&
+          event.class === 'shortcut'
+        ) {
+          const driveId = event?.referenced_by?.[0]?.id
+          if (driveId) {
+            log.info(`Shared drive ${driveId} deleted`)
+            worker?.removeSharedDrive(driveId)
+          }
+        }
+      }
+      realtime.subscribe('created', 'io.cozy.files', handleFileCreated)
+      realtime.subscribe('deleted', 'io.cozy.files', handleFileDeleted)
+      return (): void => {
+        realtime.unsubscribe('created', 'io.cozy.files', handleFileCreated)
+        realtime.unsubscribe('deleted', 'io.cozy.files', handleFileDeleted)
+      }
+    }, [client, worker])
+
     useEffect(() => {
       if (!client) return
 
       const doAsync = async (): Promise<void> => {
         // Cleanup any remaining local data
         await removeStaleLocalData()
 
+        const { data: sharedDrives } = await client
+          .collection('io.cozy.sharings')
+          .fetchSharedDrives()
+
+        const sharedDriveIds: string[] = sharedDrives.map(
+          (drive: { id: string }) => drive.id
```

Comment:
be careful with `xxx.id` if `xxx` is an io.cozy.doc we should rely on `._id` instead because `.id` is not always defined (specially with realtime)